src/main/java/com/minecolonies/coremod/MineColonies.java

```diff
@@ -170,4 +175,56 @@ public void serverLoad(final FMLServerStartingEvent event)
         // register server commands
         event.registerServerCommand(new CommandEntryPoint());
     }
+
+    /**
+     * this checks that you are not in liquid.  Will check for all liquids, even those from other mods
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @param sender uses the player to get the world
+     * @return isSafe true=safe false=water or lava
+     */
+    public static boolean isPositionSafe(@NotNull ICommandSender sender, BlockPos blockPos)
+    {
+        return sender.getEntityWorld().getBlockState(blockPos).getBlock() != Blocks.AIR
+                && !sender.getEntityWorld().getBlockState(blockPos).getMaterial().isLiquid()
+                && !sender.getEntityWorld().getBlockState(blockPos.up()).getMaterial().isLiquid();
+    }
+
+    /**
+     * this checks that you are not in the air or underground
+     * and if so it will look up and down for a good landing spot
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @return blockPos to be used for the TP
+     */
+    public static BlockPos findLand(final BlockPos blockPos, final World world)
```

Comment:
this is definitely not where this belongs. Please move to an appropriate util class